assets/js/types.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable camelcase */
+
+type Zones = 'n' | 'e' | 's' | 'w' | 'c' | 'm';
+
+type ZoneConfig = {
+  value: string | boolean;
+  modes: {
+    auto: boolean;
+    off: boolean;
+    headway: boolean;
+    custom: boolean;
+  };
+}
+
+type StationConfig = {
+    id: string;
+    name: string;
+    zonePositions: {
+      left: Zones[];
+      right: Zones[];
+      center: Zones[];
+    };
+    zones: {
+      n: ZoneConfig;
+      e: ZoneConfig;
+      s: ZoneConfig;
+      w: ZoneConfig;
+      c: ZoneConfig;
+      m: ZoneConfig;
+    };
+  }
+
+type SignConfig = { id?: string, mode: 'auto' | 'headway' | 'off' | 'static_text', expires?: string | null, line1?: string; line2?: string}
```

Comment:
This seems quite long. Do we have a line length limit as part of our code prettification/format checking?

Reply:
i actually don't think we do. we haven't actually set up anything for formatting besides running `eslint --fix`, which will do things like change double quotes into single quotes, but apparently not this. I thought about adding that onto this PR, but was trying to limit the diff.

Reply:
hm, I see. created an inbound task for us to think about: https://app.asana.com/0/881264583703207/1199158942990536